src/easyreflectometry/sample/assemblies/surfactant_layer.py

```diff
@@ -44,8 +45,17 @@ def __init__(
         :param conformal_roughness: Constrain the roughness to be the same for both layers, defaults to `False`.
         :param interface: Calculator interface, defaults to `None`.
         """
+        if unique_name is None:
+            unique_name = global_object.generate_unique_name(self.__class__.__name__)
+
         if tail_layer is None:
-            air = Material(0, 0, 'Air')
+            air = Material(
+                sld=0,
+                isld=0,
+                name='Air',
+                unique_name=unique_name + 'MaterialTail',
```

Comment:
Just a stylistic question: the syntax for unique names is to have a class name with `_<n>` attached. Adding `MaterialTail` etc. will look weird with the leading number `<n>`.
Maybe it would be more readable to add a leading underscore so the unique name would be
`SurfactantLayer_7_MaterialTail` rather than `SurfactantLayer_7MaterialTail`

Reply:
Using suggested syntax